packages/react-core/src/components/Button/Button.tsx

```diff
@@ -69,29 +75,68 @@ const Button: React.FunctionComponent<ButtonProps & InjectedOuiaProps> = ({
   icon = null,
   ouiaContext = null,
   ouiaId = null,
-  tabIndex = null as number,
+  tabIndex = null,
   ...props
 }: ButtonProps & InjectedOuiaProps) => {
   const Component = component as any;
   const isButtonElement = Component === 'button';
+
+  const getDefaultPreventedEvents = () =>
+    inoperableEvents.map(eventToPrevent => ({
+      [eventToPrevent]: (event: React.SyntheticEvent<HTMLButtonElement>) => {
+        event.preventDefault();
+      }
+    }));
+
+  const linkBtnEvents = () => {
+    if (!isButtonElement && inoperableEvents.includes('onKeyPress')) {
+      return {
+        onClick: (event: React.SyntheticEvent<HTMLButtonElement>) => {
+          event.preventDefault();
+        }
+      };
+    }
+  };
+
+  let hasWarnBeta = false;
+
+  if (isAriaDisabled && process.env.NODE_ENV !== 'production' && !hasWarnBeta) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'You are using a beta component feature (isAriaDisabled). These api parts are subject to change in the future.'
+    );
+    hasWarnBeta = true;
+  }
+
+  const preventedEvents = Object.assign({}, linkBtnEvents(), ...getDefaultPreventedEvents());
```

Comment:
Take it or leave it, but you could simplify this by replacing `getDefaultPreventedEvents` and `linkBtnEvents` and this `Object.assign` with a single object-building array reduce:

```js
const preventDefault = (event: React.SyntheticEvent<HTMLButtonElement>) => {
  event.preventDefault();
};
const preventedEvents = inoperableEvents.reduce((handlers, eventToPrevent) => ({
  ...handlers,
  [eventToPrevent]: preventDefault,
  ...(!isButtonElement && eventToPrevent === 'onKeyPress' && { onClick: preventDefault })
}), {});
```

Reply:
Or if the logic in the reduce is too awkward, this works too:

```js
const eventsToPrevent =
  !isButtonElement && inoperableEvents.includes('onKeyPress')
    ? [...inoperableEvents, 'onClick']
    : inoperableEvents;
const preventedEvents = eventsToPrevent.reduce(
  (handlers, eventToPrevent) => ({
    ...handlers,
    [eventToPrevent]: (event: React.SyntheticEvent<HTMLButtonElement>) => {
      event.preventDefault();
    },
  }),
  {}
);
```

Reply:
Thanks! Yea I didn't feel quite settled on this, I'll give it another round of refactoring. Thx for the snippet :)